### PR TITLE
F811, F523/F524, F841, F741 - Corrections in crystal.py

### PR DESCRIPTION
**Pull request by alex-rakowski**
F811, F523/F524, F841, F741 - Corrections

### CODE REVIEW

**Comment by alex-rakowski** [comment]
added the changes to `linter.yaml`, `.flake8` and removed `build-flake.yaml` as the GA is in a strange state in dev at the moment, so the test would fail.

File: py4DSTEM/process/diffraction/crystal.py [hunk 1]
```diff
@@ -660,7 +649,8 @@ def generate_diffraction_pattern(
             self.setup_diffraction(300e3)
 
         # Tolerance for angular tests
-        tol = 1e-6
+        # TODO should this be used somewhere ?
```
Comment (sezelt) [inline py4DSTEM/process/diffraction/crystal.py:652]: This does seem to be unused, we can delete it

File: py4DSTEM/process/diffraction/crystal.py [hunk 2]
```diff
@@ -816,7 +806,8 @@ def generate_ring_pattern(
 
         # check accelerating voltage
         if hasattr(self, "accel_voltage"):
-            accelerating_voltage = self.accel_voltage
+            # TODO should this be used somewhere ?
+            accelerating_voltage = self.accel_voltage  # noqa: F841
```
Comment (sezelt) [inline py4DSTEM/process/diffraction/crystal.py:810]: This is also unused, can be deleted.

**Review by sezelt — CHANGES_REQUESTED** [review]
The changes look good, you can delete those two unused variables in `crystal.py` and then I think this is ready to go.

**Comment by sezelt** [comment]
This does seem to overlap with #563 a bit, so maybe they should be combined? Probably the only potential conflict is if they make different changes to the flake8 file

File: py4DSTEM/process/diffraction/crystal.py (hunk 1 above)
Comment (alex-rakowski) [reply py4DSTEM/process/diffraction/crystal.py:652]: removed

File: py4DSTEM/process/diffraction/crystal.py (hunk 2 above)
Comment (alex-rakowski) [reply py4DSTEM/process/diffraction/crystal.py:810]: removed

File: py4DSTEM/process/diffraction/crystal.py [hunk 3]
```diff
@@ -815,9 +801,7 @@ def generate_ring_pattern(
             )
 
         # check accelerating voltage
-        if hasattr(self, "accel_voltage"):
-            accelerating_voltage = self.accel_voltage
-        else:
+        if not hasattr(self, "accel_voltage"):
```
Comment (alex-rakowski) [inline py4DSTEM/process/diffraction/crystal.py:804]: @sezelt deleted the var, but this should preserve the functionality

**Review by sezelt — APPROVED** [review]

**Comment by sezelt** [comment]
I'm approving this since all the changes it makes are acceptable, but I'll hold off on merging so that we can decide how to reconcile with #563

**Review by bsavitzky — CHANGES_REQUESTED** [review]
See comment in-thread

**Comment by bsavitzky** [comment]
Hey @alex-rakowski - 

This PR is redundant / in conflict with PR#563.  All file lines changed here other than those in crystal.py have already been modified, often quite differently, in that PR.  Can we sync up?

Ultimately this should be a single PR, otherwise we're just making more work for ourselves!

**Comment by sezelt** [comment]
I've merged this with #560

**Pull request closed without merging**